symbolic-il2cpp/src/usym.rs

```diff
@@ -360,56 +394,91 @@ impl<'a> UsymSymbols<'a> {
+        let nsymbol_offset = raw.native_symbol.try_into().unwrap();
+        let native_symbol = self.get_string(nsymbol_offset)?;
+        let native_symbol_bytes = self.get_string_bytes(nsymbol_offset)?;
+
+        let nfilename_offset = raw.native_file.try_into().unwrap();
+        let native_file = self.get_string(nfilename_offset)?;
+        let native_file_info = {
+            let file_bytes = self.get_string_bytes(nfilename_offset)?;
+            // implementation blatantly stolen from FileInfo::from_path because it's only visible to
+            // the debuginfo crate
+            let (dir, name) = symbolic_common::split_path_bytes(file_bytes);
+            FileInfo {
+                name,
+                dir: dir.unwrap_or_default(),
+            }
+        };
 
-        let managed_symbol = self.get_string(raw.managed_symbol.try_into().unwrap())?;
-        let managed_symbol = match managed_symbol.is_empty() {
-            true => None,
-            false => Some(managed_symbol),
+        let msymbol_offset = raw.managed_symbol.try_into().unwrap();
+        let managed_symbol = self.get_string(msymbol_offset)?;
+        let managed_symbol = if managed_symbol.is_empty() {
+            None
+        } else {
+            Some(managed_symbol)
         };
         if managed_symbol.is_none() && raw.managed_symbol > 0 {
             println!("A managed symbol with a >0 offset into the string table points to an empty string. We normally expect empty strings to have an offset of 0.");
             println!("Native entry: {}::{}", native_file, native_symbol);
         }
-
-        let managed_file = self.get_string(raw.managed_file.try_into().unwrap())?;
-        let managed_file = match managed_file.is_empty() {
+        let mfilename_offset = raw.managed_file.try_into().unwrap();
+        let managed_file = self.get_string(mfilename_offset)?;
+        let managed_file_info = match managed_file.is_empty() {
             true => None,
-            false => Some(managed_file),
+            false => {
+                let file_bytes = self.get_string_bytes(mfilename_offset)?;
+                // implementation blatantly stolen from FileInfo::from_path because it's only visible to
+                // the debuginfo crate
+                let (dir, name) = symbolic_common::split_path_bytes(file_bytes);
+                Some(FileInfo {
+                    name,
+                    dir: dir.unwrap_or_default(),
+                })
+            }
         };
-        if managed_file.is_none() && raw.managed_file > 0 {
-            println!("A managed file with a >0 offset into the string table points to an empty string. We normally expect empty strings to have an offset of 0.");
+        if managed_file.is_empty() && raw.managed_file > 0 {
+            println!("A managed file name with a >0 offset into the string table points to an empty string. We normally expect empty strings to have an offset of 0.");
```

Comment:
The record is, or well should still be usable even if the filename or symbol are empty - as long as at least one of those contain a usable string I think we should be able to use it. I wasn't able to find any evidence of a logging crate in any of the `Cargo.toml` in the repository so I've just opted to comment these out for the time being. I'll follow up with a PR that turns these into proper warnings once I figure that out.

Which `Cargo.toml` and logging crate were you thinking of?

Reply:
I think the `log` crate is the default logging facade in rust, which afaik is also usable by projects using `tracing` so that's probably just fine.

`Cargo.toml` can only be the one of `symbolic-il2cpp` if you want to use logging...?  I may have missed what the question was here.